thesslstore_module.php

```diff
@@ -66,6 +66,276 @@ public function getAuthors() {
+            Loader::loadComponents($this, ['Record']);
+        }
+        Loader::loadModels($this,  ['CronTasks']);
+
+        $cron_tasks = $this->getCronTasks();
+
+        if ($last_instance) {
+            // Remove the cron tasks
+            foreach ($cron_tasks as $task) {
+                $cron_task = $this->CronTasks->getByKey($task['key'], $task['dir'], $task['task_type']);
+                if ($cron_task) {
+                    $this->CronTasks->deleteTask($cron_task->id, 'module', $task['dir']);
+                }
+            }
+        }
+
+        // Remove individual cron task runs
+        foreach ($cron_tasks as $task) {
+            $cron_task_run = $this->CronTasks
+                ->getTaskRunByKey($task['key'], $task['dir'], false, $task['task_type']);
+            if ($cron_task_run) {
+                $this->CronTasks->deleteTaskRun($cron_task_run->task_run_id);
+            }
+        }
+    }
+
+    /**
+     * Performs migration of data from $current_version (the current installed version)
+     * to the given file set version. Sets Input errors on failure, preventing
+     * the module from being upgraded.
+     *
+     * @param string $current_version The current installed version of this module
+     */
+    public function upgrade($current_version)
+    {
+		// Upgrade if possible
+		if (version_compare($current_version, "1.7.0", "<")) {
+			$this->addCronTasks($this->getCronTasks());
+		}
+    }
+
+    /**
+     * Runs the cron task identified by the key used to create the cron task
+     *
+     * @param string $key The key used to create the cron task
+     * @see CronTasks::add()
+     */
+    public function cron($key)
+    {
+        if ($key == 'tss_order_sync') {
+            $this->orderSynchronization();
+        }
+    }
+
+    /**
+     * Retrieves cron tasks available to this module along with their default values
+     *
+     * @return array A list of cron tasks
+     */
+    private function getCronTasks()
+    {
+        return [
+            [
+                'key' => 'tss_order_sync',
+                'task_type' => 'module',
+                'dir' => 'thesslstore_module',
+                'name' => Language::_('ThesslstoreModule.getCronTasks.tss_order_sync_name', true),
+                'description' => Language::_('ThesslstoreModule.getCronTasks.tss_order_sync_desc', true),
+                'type' => 'time',
+                'type_value' => '00:00:00',
+                'enabled' => 1
+            ]
+        ];
+    }
+
+    /**
+     * Attempts to add new cron tasks for this module
+     *
+     * @param array $tasks A list of cron tasks to add
+     */
+    private function addCronTasks(array $tasks)
+    {
+        Loader::loadModels($this, ['CronTasks']);
+        foreach ($tasks as $task) {
+            $task_id = $this->CronTasks->add($task);
+
+            if (!$task_id) {
+                $cron_task = $this->CronTasks->getByKey($task['key'], $task['dir'], $task['type']);
```

Comment:
Shouldn't the third argument to ::getByKey be the task type, not the task run type?